cloudbuild.yaml

```diff
@@ -189,6 +189,31 @@ steps:
       - check-cloud-marketplace-tools
       - push-smoke-test
 
+  - id: logs-jetstack-secure-gcm
+    name: gcr.io/cloud-builders/gcloud
+    entrypoint: bash
+    args:
+      - -euc
+      - |
+        until kubectl get ns -oname 2>/dev/null | grep apptest-; do
+          sleep 10
+        done
+        ns=$(kubectl get ns -oname  | grep apptest- | cut -d/ -f2)
+
+        set -x
+        while kubectl get ns "$ns"; do
+          kubectl logs \
+            --namespace "$ns" \
+            --follow \
+            --all-containers \
+            --prefix \
+            --selector app.kubernetes.io/name=jetstack-secure-gcm || true
+          sleep 10
+        done
+    waitFor:
+      - check-cloud-marketplace-tools
+      - push-deployer
```

Comment:
This is a new step which collects the logs of all the JSP-GCM components for easier debugging when the smoke-tests fail.
It gets the logs of all the containers of all the pods having label `app.kubernetes.io/name=jetstack-secure-gcm` and prefixes each log line with the ns/pod/container

This helped me figure out that I'd forgotten to add the missing `Secret > Patch` RBAC to the Schema.yaml file.

Reply:
Here's what the output looks like: https://console.cloud.google.com/cloud-build/builds/e16dbf32-b48b-4153-8c86-91040b9d822c;step=13?project=885059085598

Reply:
Thanks, very useful indeed!